sieve_of_eratosthenes/sieveOfEratosthenes.js

```diff
@@ -0,0 +1,31 @@
+function sieveOfEratosthenes (n) {
+	/*
+	* Calculates prime numbers till a number n
+	* :param n: Number upto which to calculate primes
+	* :return: A boolean list contaning only primes
+	*/
+	let primes = new Array(n + 1);
+	primes.fill(true);  // set all as true initially
+	primes[0] = primes[1] = false;  // Handling case for 0 and 1
+	let sqrtn = Math.ceil(Math.sqrt(n));
+	for (let i = 2; i <= sqrtn; i++) {
+		if (primes[i]) {
+			for (let j = 2 * i; j <= n; j += i) {
+				primes[j] = false;
+			}
+		}
+	}
+	return primes;
+}
+
+function main () {
+	let n = 319; // number till where we wish to find primes
+	let primes = sieveOfEratosthenes(n);
+	for (let i = 1; i <= n; i++) {
```

Comment:
> Then why did we deal with base case of 0 and 1, we could've started checking from 2 itself.

We dealt with them because **our function should be correct**. Now if someone checks `primes[1]`, it should be false and not something else. 

In the `main` method, we are listing the primes. Everyone knows 1 is not prime, so it's ok to start with 2.